input4linux-0.8.0-rc2.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+set -euo pipefail
+
+# Configuration
+TEST_MODE="${TEST_MODE:-true}"
+URL="https://github.com/worklouder/input-releases/releases/download/v0.8.0-rc.2/input-Setup-0.8.0-rc.2.exe"
+FILENAME="input-Setup-0.8.0-rc.2.exe"
+DOWNLOAD_DIR="./input_download"
+EXTRACT_DIR="./input_extracted"
+REBUILD_DIR="./input_rebuild"
+APP_64_EXTRACT_DIR="$REBUILD_DIR/app-64"
+FINAL_APP_DIR="./input-app"
+ELECTRON_VERSION="29.2.0"
+
+# Ensure required tools are available
+for cmd in curl 7z asar npm; do
+    if ! command -v "$cmd" &>/dev/null; then
+        echo "Missing required command: $cmd"
+        [[ "$TEST_MODE" == true ]] && echo "Test mode enabled, continuing..." || exit 1
+    fi
+done
+
+# Error handling
+handle_error() {
+    local message="$1"
+    if [[ "$TEST_MODE" == true ]]; then
+        echo "$message (ignored due to TEST_MODE)"
+    else
+        echo "$message"
+        exit 1
+    fi
+}
+
+# Prepare directories
+mkdir -p "$DOWNLOAD_DIR" "$EXTRACT_DIR" "$REBUILD_DIR" "$APP_64_EXTRACT_DIR"
+
+# Download EXE
+echo "Downloading $FILENAME..."
```

Comment:
Instead of redownloading everytime, we can check to see if the file exists and is valid to save time and bandwidth (not that its a big download). Something like this:
```
if [[ -f "$DOWNLOAD_DIR/$FILENAME" ]]; then
    echo "$FILENAME already exists, skipping download"
else
    curl -L "$URL" -o "$DOWNLOAD_DIR/$FILENAME" || handle_error "Download failed"
fi
```

Reply:
The issue is that $DOWNLOAD_DIR doesn't exist yet when you're trying to check for or write the file. curl won’t automatically create parent directories, and -f doesn’t implicitly create them either.

```
mkdir -p "$DOWNLOAD_DIR"

if [[ -f "$DOWNLOAD_DIR/$FILENAME" ]]; then
    echo "$FILENAME already exists, skipping download"
else
    curl -L "$URL" -o "$DOWNLOAD_DIR/$FILENAME" || handle_error "Download failed"
fi
```